### DISCUSSION TITLE
Set permissions programmatically

### DISCUSSION THREAD

**Discussion by davidbrochart** [fief-dev/fief#223] [Q&A] [answered]
I am trying to set user's permissions programmatically using the admin API, through the [POST /users/{id}/permissions](https://example.fief.dev/admin/api/docs#/Users/users_create_permission_users__id__permissions_post), but it looks like the body containing the ID of the permission must be a UUID, while I was expecting it to be the "codename", e.g.:
```json
{
  "id": "contents:read"
}
```
Am I missing something?

**Comment by filipagh** [comment, marked as answer]
you need load permissions and use its id https://example.fief.dev/admin/api/docs#/Permissions/permissions_list_permissions__get

**Reply by davidbrochart** [reply]
Nice, thanks a lot @filipagh !
